### PR TITLE
Move the Installer-related code into a different package

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes https://dev.azure.com/msazure/AzureRedHatOpenShift/_workitems/edit/14625890

### What this PR does / why we need it:

Moves the code that will later be pulled into/replaced by the standalone installer into a separate package for clarity. Contains no real logic changes.

### Test plan for issue:

Unit tests & E2E. 

### Is there any documentation that needs to be updated for this PR?
Reorganisation, N/A

### CODE REVIEW

**Review by s-amann — DISMISSED** [review]

File: pkg/cluster/install.go [hunk 1]
```diff
@@ -124,55 +122,49 @@ func (m *manager) Update(ctx context.Context) error {
...
 			steps.Action(m.createAPIServerPrivateEndpoint),
 			steps.Action(m.createCertificates),
+
+			// Run installer. For M5/M6 we will persist the graph inside the
+			// installer code since it's easier, but in the future, this data
+			// should be collected from Hive's outputs where needed.
+			steps.Action(m.callInstaller),
+
+			//steps.Action(func(ctx context.Context) error {
+			//  return m.persistGraph(ctx, g)
```
[the remark this answers is not in the document]
Comment (darthhexx) [reply pkg/cluster/install.go:161]: I agree, this is my only nit as well, otherwise LGTM. Can we remove this @hawkowl?

**Review by darthhexx — DISMISSED** [review]

File: pkg/installer/condition.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+// condition functions should return an error only if it's not retryable
+// if a condition function encounters a retryable error it should return false, nil.
+
+func (m *manager) bootstrapConfigMapReady(ctx context.Context) (bool, error) {
+	cm, err := m.kubernetescli.CoreV1().ConfigMaps("kube-system").Get(ctx, "bootstrap", metav1.GetOptions{})
+	if err != nil && m.env.IsLocalDevelopmentMode() {
+		m.log.Printf("bootstrapConfigMapReady condition error %s", err)
+	}
+	return err == nil && cm.Data["status"] == "complete", nil
+}
```
Comment (AldoFusterTurpin) [inline pkg/installer/condition.go:21]: IMHO we could break the return line and add names to the return values to make the function a bit cleaner and explicit, what about something like:
```go
func (m *manager) bootstrapConfigMapReady(ctx context.Context) (successfullyCompleted bool, err error) {
	cm, err := m.kubernetescli.CoreV1().ConfigMaps("kube-system").Get(ctx, "bootstrap", metav1.GetOptions{})
	if err != nil && m.env.IsLocalDevelopmentMode() {
		m.log.Printf("bootstrapConfigMapReady condition error %s", err)
	}

	statusCompleted := cm.Data["status"] == "complete"

	return err == nil && statusCompleted, nil
}
```

?

**Review by AldoFusterTurpin — COMMENTED** [review]
@hawkowl Great! I have just added a couple of suggestions that I think simplify a bit the code.

File: pkg/cluster/install.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hawkowl) [reply pkg/cluster/install.go]: Done

**Review by darthhexx — APPROVED** [review]
LGTM

**Review by s-amann — APPROVED** [review]
LGTM

**Pull request merged**